app/views/session_timeout/_ping.js.erb

```diff
@@ -20,13 +21,28 @@ keepaliveEl.addEventListener('click', keepalive, false);
 var pingTimeout;
 var countdownInterval;
 
+function notifyNewRelic(error, actionName) {
+  if (typeof newrelic !== 'object') {
+    return;
+  }
+
+  newrelic.addPageAction('Session Auto Logout', {
+    duration_ms: new Date() - initialTime,
+    action_name: actionName,
+    error: error.message
+  });
+}
+
 function ping() {
   var request = new XMLHttpRequest();
   request.open('GET', '/active', true);
 
   request.onload = function() {
-    if (request.status >= 200 && request.status < 400) {
+    try {
```

Comment:
My thinking was, well, we're adding a try/catch, might as well not even have a guard here.

Another thing we could do (looking for feedback here) is to pass in `request` to the `notifyNewRelic` function so we could get info on the status codes we're seeing

Reply:
I do kinda like the idea of passing along status codes, or at least having more informed details about why a JSON parse failed, not just that it failed. I think this could also extend the original bug itself: If we're parsing HTML from a redirect (like CSRF token invalidation), maybe we should be instead be returning more meaningful responses from the server (like 4xx, maybe with a response text, maybe responding to JSON as JSON).

Reply:
Added `request_status` in 49eda944e073e911bf22d1ae185ad6f5d25591fa